### PR TITLE
[CAS-283] Tap to retry failed message

### PR DESCRIPTION
**Pull request by zsmb13**
https://stream-io.atlassian.net/browse/CAS-283

Implements the missing "click to retry" functionality for failed messages.

Replaces #671

### CODE REVIEW

File: library/src/main/java/com/getstream/sdk/chat/ChatSocketListener.kt [hunk 1]
```diff
@@ -141,7 +141,6 @@ internal class ChatSocketListener(
             is ConnectingEvent,
             is DisconnectedEvent,
             is ErrorEvent,
-            is ChannelUpdatedByUserEvent,
```
Comment (zsmb13) [inline library/src/main/java/com/getstream/sdk/chat/ChatSocketListener.kt:144]: This was a duplicate type in this branch of the `when` statement, just cleaned up a warning by removing it

File: library/src/main/java/com/getstream/sdk/chat/adapter/viewholder/message/configurators/MessageTextConfigurator.kt [hunk 2]
```diff
@@ -160,10 +161,13 @@ internal class MessageTextConfigurator(
...
     ) {
         binding.tvText.setOnClickListener {
-            if (message.isFailed() && !ChatClient.instance().isSocketConnected()) {
-                return@setOnClickListener
+            if (message.isFailed()) {
+                if (ChatClient.instance().isSocketConnected()) {
+                    messageRetryListener.onRetryMessage(message)
+                }
+            } else {
+                messageClickListener.onMessageClick(message)
```
Comment (JcMinarro) [inline library/src/main/java/com/getstream/sdk/chat/adapter/viewholder/message/configurators/MessageTextConfigurator.kt:169]: The connection logic should be out of the view
```suggestion
            if (message.isFailed()) {
                    messageRetryListener.onRetryMessage(message)
            } else {
                messageClickListener.onMessageClick(message)
```

On the other hand, why don't we attach the listener only on the "Message wasn't sent, try again" text?

Comment (zsmb13) [reply library/src/main/java/com/getstream/sdk/chat/adapter/viewholder/message/configurators/MessageTextConfigurator.kt]: I removed the network check. Tested it, and the UX is pretty okay even if we don't have a connection when we perform the retry.

Also improved the condition, it now matches the condition for displaying the tap-to-retry text exactly.

**Review by JcMinarro — APPROVED** [review]
LGTM

**Pull request merged**